pkg/controller/bastion/actuator.go

```diff
@@ -0,0 +1,221 @@
+		return nil, err
+	}
+	return computeclient.FindServersByName(name)
+}
+
+func createBastionInstance(openstackClientFactory openstackclient.Factory, parameters servers.CreateOpts) (*servers.Server, error) {
+	computeclient, err := openstackClientFactory.Compute()
+	if err != nil {
+		return nil, err
+	}
+	return computeclient.CreateServer(parameters)
+}
+
+func deleteBastionInstance(openstackClientFactory openstackclient.Factory, id string) error {
+	computeclient, err := openstackClientFactory.Compute()
+	if err != nil {
+		return err
+	}
+	return computeclient.DeleteServer(id)
+}
+
+// GetIPs return privateip, publicip
+func GetIPs(s *servers.Server, opt *Options) (string, string, error) {
+	var privateIP, publicIp string
+
+	type InstanceNic struct {
+		MacAddr string `json:"OS-EXT-IPS-MAC:mac_addr"`
+		Version int    `json:"version"`
+		Addr    string `json:"addr"`
+		Type    string `json:"OS-EXT-IPS:type"`
+	}
+
+	instanceNic := []InstanceNic{}
+
+	if len(s.Addresses) == 0 {
+		return "", "", fmt.Errorf("NIC not ready yet")
+	}
+
+	bytes, err := json.Marshal(s.Addresses[opt.ShootName])
+	if err != nil {
+		return "", "", err
+	}
+	err = json.Unmarshal(bytes, &instanceNic)
+	if err != nil {
+		return "", "", err
+	}
+
+	for i, v := range instanceNic {
+		if v.Type == "fixed" {
+			privateIP = instanceNic[i].Addr
+		} else {
+			publicIp = instanceNic[i].Addr
+		}
+	}
+
+	return privateIP, publicIp, nil
+}
+
+func createFloatingIP(openstackClientFactory openstackclient.Factory, parameters floatingips.CreateOpts) (*floatingips.FloatingIP, error) {
+	client, err := openstackClientFactory.Networking()
+	if err != nil {
+		return nil, err
+	}
+	return client.CreateFloatingIP(parameters)
+}
+
+func deleteFloatingIP(openstackClientFactory openstackclient.Factory, id string) error {
+	client, err := openstackClientFactory.Networking()
+	if err != nil {
+		return err
+	}
+	return client.DeleteFloatingIP(id)
+}
+
+func associateFIPWithInstance(openstackClientFactory openstackclient.Factory, id string, parameter computefip.AssociateOpts) error {
+	client, err := openstackClientFactory.Compute()
+	if err != nil {
+		return err
+	}
+	return client.AssociateFIPWithInstance(id, parameter)
+}
+
+func findFloatingIDByInstanceID(openstackClientFactory openstackclient.Factory, id string) (string, error) {
+	client, err := openstackClientFactory.Compute()
+	if err != nil {
+		return "", err
+	}
+	return client.FindFloatingIDByInstanceID(id)
+}
+
+func getExternalNetworkInfoByName(openstackClientFactory openstackclient.Factory, name string) ([]networks.Network, error) {
+	client, err := openstackClientFactory.Networking()
+	if err != nil {
+		return nil, err
+	}
+	return client.GetExternalNetworkInfoByName(name)
+}
+
+func getFipByName(openstackClientFactory openstackclient.Factory, name string) ([]floatingips.FloatingIP, error) {
+	client, err := openstackClientFactory.Networking()
+	if err != nil {
+		return nil, err
+	}
+	return client.GetFipbyName(name)
```

Comment:
please search for `byName` (match case) and you will find all places that need to be fixed -> rename to `GetFipByName`

Reply:
fixed in 48ce533